### DISCUSSION TITLE
How to programmatically set the properties "liveliness_protection_kind" & "discovery_protection_kind" for DDS participants [13813]

### DISCUSSION THREAD

**Discussion by JaganKumar-TRM** [eProsima/Fast-DDS#2404] [Q&A] [answered]
Without using the governance file, How to programmatically set the properties "**liveliness_protection_kind**" & "**discovery_protection_kind**" for DDS participants?.

Because there is no documentation for that on the “https://fast-dds.docs.eprosima.com” site.

**Comment by EduPonz** [comment, marked as answer]
Hi @JaganKumar-TRM ,

Unfortunately, what you intend to achieve is not possible. This is because these properties are configured using the [Domain Governance Document](https://fast-dds.docs.eprosima.com/en/latest/fastdds/security/access_control_plugin/access_control_plugin.html#domain-governance-document), which is a CA signed document. Configuring them programmatically would allow for configuring without proper clearance from the CA.

**Reply by davebranton** [reply]
Given this, why is it nevertheless possible to set the message encryption programatically, via "rtps_protection_kind"? 

What security issues are created if we are unable to set either of these properties?

**Reply by EduPonz** [reply]
Hi @davebranton ,

All the security related properties are indeed being removed in Fast DDS v2.6.0 due mid-March 2022. All of them are already configurable using the signed Domain Governance Document, so applications are already capable of configuring them according to the [OMG DDS Security](https://www.omg.org/spec/DDS-SECURITY/1.1/About-DDS-SECURITY/) standard.
